recognition/include/pcl/recognition/impl/linemod/line_rgbd.hpp

```diff
@@ -167,36 +168,20 @@ pcl::LineRGBD<PointXYZT, PointRGBT>::computeBoundingBoxAndCenterTemplatePointClo
   bb.x = bb.y = bb.z = std::numeric_limits<float>::max ();
   bb.width = bb.height = bb.depth = 0.0f;
 
-  Eigen::Vector3f geometric_center = Eigen::Vector3f::Zero ();
-  Eigen::Vector3f min_pos, max_pos;
+  Eigen::Vector4f geometric_center = Eigen::Vector4f::Zero ();
+  Eigen::Vector4f min_pos, max_pos;
   min_pos.fill (std::numeric_limits<float>::max ());
   max_pos.fill (std::numeric_limits<float>::lowest ());
-  std::size_t counter = 0;
-  for (std::size_t j = 0; j < template_point_cloud.size (); ++j) 
-  {
-    const PointXYZRGBA & p = template_point_cloud.points[j];
-
-    if (!isXYZFinite(p))
-      continue;
-
-    for (std::size_t row=0; row<3; ++row)
-    {
-      min_pos[row] = std::min (min_pos[row], p.getVector3fMap ()[row]);
-      max_pos[row] = std::max (max_pos[row], p.getVector3fMap ()[row]);
-    }
-
-    geometric_center += p.getVector3fMap ();
-
-    ++counter;
-  }
-  geometric_center /= static_cast<float> (counter);
+  
+  pcl::getMinMax3D(template_point_cloud, min_pos, max_pos);
```

Comment:
That's not what I meant. You've effectively split up cache misses into 2 loops. I meant copy the code (the use of arrays for min/max) from there

Reply:
@kunaltyagi Is the advantage of looping over the array elements vs using Eigen's min/max method significant?

Reply:
Eigen almost always does vectorization at O2 while GCC and clang do that at O3. This has a cascading effect on the outer loop which also has a better chance of getting vectorised